File: 2017-05-09/ContentConsumer/app/src/main/java/br/ufpe/cin/if1001/contentconsumer/ContentProviderContract.java
package br.ufpe.cin.if1001.contentconsumer;

import android.content.ContentResolver;
import android.net.Uri;

public final class ContentProviderContract {
    public static final String _ID = "_id";
    public static final String NOME = "nome";
    public static final String CPF = "cpf";
    public static final String EMAIL = "email";
    public static final String PESSOA_TABLE = "pessoa";

    private static final Uri BASE_ARRAY_URI = Uri.parse("content://br.ufpe.cin.if1001.arrayprovider/");
    //URI para tabela
    public static final Uri CONTENT_ARRAY_URI = Uri.withAppendedPath(BASE_ARRAY_URI, PESSOA_TABLE);

    private static final Uri BASE_DB_URI = Uri.parse("content://br.ufpe.cin.if1001.dbprovider/");
    //URI para tabela
    public static final Uri CONTENT_DB_URI = Uri.withAppendedPath(BASE_DB_URI, PESSOA_TABLE);

    private static final Uri BASE_AULA_URI = Uri.parse("content://br.ufpe.cin.if1001.aulaprovider/");
    //URI para tabela
    public static final Uri CONTENT_AULA_URI = Uri.withAppendedPath(BASE_AULA_URI, PESSOA_TABLE);

    // Mime type para colecao de itens
    public static final String CONTENT_DIR_TYPE = ContentResolver.CURSOR_DIR_BASE_TYPE + "/PessoaContentProvider.data.text";

    // Mime type para um item especifico
    public static final String CONTENT_ITEM_TYPE = ContentResolver.CURSOR_ITEM_BASE_TYPE + "/PessoaContentProvider.data.text";

    // Todas as colunas do content provider
    public static final String[] ALL_COLUMNS = { _ID, NOME, CPF, EMAIL };

}

File: 2019-05-13/ContentProvider/app/src/main/java/br/ufpe/cin/android/contentprovider/ContentProviderContract.kt
package br.ufpe.cin.android.contentprovider

import android.content.ContentResolver
import android.net.Uri

object ContentProviderContract {
    val _ID = "_id"
    val NOME = "nome"
    val CPF = "cpf"
    val EMAIL = "email"
    val MEDIA = "media"
    val PESSOA_TABLE = "pessoa"


    val STATE_CODE = "uf"
    val STATE_NAME = "name"
    val ESTADO_COLUMNS = arrayOf(_ID, STATE_CODE, STATE_NAME)
    val ESTADO_TABLE = "estado"


    val BASE_LIST_URI = Uri.parse("content://br.ufpe.cin.android.listprovider/")
    //URI para tabela
    val CONTENT_LIST_URI = Uri.withAppendedPath(BASE_LIST_URI, PESSOA_TABLE)

    val BASE_SQL_URI = Uri.parse("content://br.ufpe.cin.android.sqlprovider/")
    //URI para tabela
    val CONTENT_ESTADOS_URI = Uri.withAppendedPath(BASE_SQL_URI, ESTADO_TABLE)

    // Mime type para colecao de itens
    val CONTENT_DIR_TYPE = ContentResolver.CURSOR_DIR_BASE_TYPE + "/PessoaContentProvider.data.text"

    // Mime type para um item especifico
    val CONTENT_ITEM_TYPE = ContentResolver.CURSOR_ITEM_BASE_TYPE + "/PessoaContentProvider.data.text"

    // Todas as colunas do content provider
    val ALL_COLUMNS = arrayOf(_ID, NOME, CPF, EMAIL, MEDIA)
}
